lib/taskana-core-test/src/test/java/acceptance/task/requestchanges/RequestChangesWithBeforeSpiAccTest.java

```diff
@@ -0,0 +1,313 @@
+      this.taskanaEngine = taskanaEngine;
+    }
+
+    @Override
+    public Task beforeRequestChanges(Task task) throws Exception {
+      Classification newClassification =
+          defaultTestClassification().buildAndStore(taskanaEngine.getClassificationService());
+
+      task.setClassificationKey(newClassification.getKey());
+      task = taskanaEngine.getTaskService().updateTask(task);
+
+      newClassification.setName(SPI_CLASSIFICATION_NAME);
+      taskanaEngine.getClassificationService().updateClassification(newClassification);
+
+      return task;
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = BeforeRequestChangesProvider.class,
+      serviceProviders = TaskModifier.class)
+  class SpiModifiesTask {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      Task result = taskService.requestChanges(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3))
+          .isEqualTo(TaskModifier.NEW_CUSTOM_3_VALUE);
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnPersistentModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      taskService.requestChanges(task.getId());
+      Task result = taskService.getTask(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3))
+          .isEqualTo(TaskModifier.NEW_CUSTOM_3_VALUE);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = BeforeRequestChangesProvider.class,
+      serviceProviders = SetTaskOwner.class)
+  class SpiSetsValuesWhichGetOverridenByTaskana {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_OverrideOwner_When_SpiModifiesOwner() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      taskService.forceRequestChanges(task.getId());
+      Task result = taskService.getTask(task.getId());
+
+      assertThat(result.getOwner()).isNull();
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ThrowError_When_SpiModifiesOwner() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      ThrowingCallable call = () -> taskService.requestChanges(task.getId());
+
+      InvalidOwnerException ex = catchThrowableOfType(call, InvalidOwnerException.class);
+      assertThat(ex.getTaskId()).isEqualTo(task.getId());
+      assertThat(ex.getCurrentUserId()).isEqualTo("user-1-1");
+    }
+
+    class SetTaskOwner implements BeforeRequestChangesProvider {
```

Comment:
Why is this SPI not placed with the other SPIs?

Reply:
Because this SPI is only needed in the enclosing testclass. You prefer that all SPI's are located in close proximity?